boa_engine/src/builtins/weak/weak_ref.rs

```diff
@@ -159,7 +159,12 @@ mod tests {
                 "#},
                 |v, _| v.is_object(),
             ),
-            TestAction::inspect_context(|_| boa_gc::force_collect()),
+            TestAction::inspect_context(|context| {
+                context.clear_kept_objects();
+                boa_gc::force_collect();
```

Comment:
Why do we need to run this three times?

Reply:
This is a leftover from my testing, I think there is a bug in the way we collect weak gc pointers, from my testing when a `WeakRef` instance has reference to a shape, it's not being collected. Even with multiple `boa_gc::force_collect()` are called one after the other, the only way I got it to collect the weak ref was by creating a huge number of objects `for (let i =0; i < 10000; ++i) { ({}) }` this is why the `Weak` builtin variants don't have shapes on their instances.

Reply:
Do we have an issue open about this? If not, I think we should open one, to debug and fix it if need it :)

Reply:
I tested it again and it seem to work fine, strange :confused: